openedx/core/djangoapps/content_libraries/signal_handlers.py

```diff
@@ -0,0 +1,43 @@
+"""
+Content library signal handlers.
+"""
+
+import logging
+
+from django.dispatch import receiver
+
+from lms.djangoapps.grades.api import signals as grades_signals
+
+from .models import LtiGradedResource
+
+
+log = logging.getLogger(__name__)
+
+
+@receiver(grades_signals.PROBLEM_WEIGHTED_SCORE_CHANGED)
+def score_changed_handler(sender, **kwargs):  # pylint: disable=unused-argument
+    """
+    Match the score event to an LTI resource and update.
+    """
+
+    modified = kwargs.get('modified')
+    usage_id = kwargs.get('usage_id')
+    user_id = kwargs.get('user_id')
+    weighted_earned = kwargs.get('weighted_earned')
+    weighted_possible = kwargs.get('weighted_possible')
+
+    if None in (modified, usage_id, user_id, weighted_earned, weighted_possible):
+        log.error("LTI 1.3: Score Signal: Missing a required parameters, "
+                  "ignoring: kwargs=%s", kwargs)
+        return
+    try:
+        resource = LtiGradedResource.objects.get_from_user_id(
+            user_id, usage_key=usage_id
+        )
+    except LtiGradedResource.DoesNotExist:
+        log.error("LTI 1.3: Score Signal: Unknown resource, ignoring: kwargs=%s",
+                  kwargs)
```

Comment:
Isn't this handler going to be called for every single problem grade event on the entire platform, including hundreds of thousands of interactions with non-library non-LTI content? If so, we definitely don't want to log anything in the (extremely common) case that there is no `LtiGradedResource` matching the event.

This will also add 1-2 additional database lookups to every single problem submit event, so to make it more performant I think you should at least add a short-circuit line that will return early if the `usage_id` is not a content libraries v2 identifier.

It would also be good to have a feature flag that disables the LTI1.3 library support entirely, including this signal handler - I'm going to mention that as a top level comment.